src/main/java/uk/gov/hmcts/ccd/data/casedetails/search/SortOrderQueryBuilder.java

```diff
@@ -1,37 +1,55 @@
 package uk.gov.hmcts.ccd.data.casedetails.search;
 
 import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Component;
 import uk.gov.hmcts.ccd.data.casedetails.search.MetaData.CaseField;
 
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 @Component
 public class SortOrderQueryBuilder {
 
+    private static final Logger LOG = LoggerFactory.getLogger(SortOrderQueryBuilder.class);
+
     private static final String DATA_FIELD = "data";
     private static final String CREATED_DATE = "created_date";
     private static final String SPACE = " ";
     private static final String COMMA = ",";
+    private static final String SPECIAL_CHARS_REGEXP = "[\\s\\\\\\/,;\\)\\('\"`]";
+    private static final Pattern SPECIAL_CHARS_PATTERN = Pattern.compile(SPECIAL_CHARS_REGEXP);
 
     public String buildSortOrderClause(MetaData metaData) {
         StringBuilder sb = new StringBuilder();
         metaData.getSortOrderFields().forEach(sortOrderField -> {
-            if (sortOrderField.isMetadata()) {
-                sb.append(getMataFieldName(sortOrderField.getCaseFieldId()));
-            } else {
-                sb.append(convertFieldNameToJSONBsqlFormat(sortOrderField.getCaseFieldId()));
+            Matcher matcher = SPECIAL_CHARS_PATTERN.matcher(sortOrderField.getCaseFieldId());
+            if (!matcher.find()) {
+                if (sortOrderField.isMetadata()) {
+                    sb.append(getMataFieldName(sortOrderField.getCaseFieldId()));
+                } else {
+                    sb.append(convertFieldNameToJSONBsqlFormat(sortOrderField.getCaseFieldId()));
+                }
+                sb.append(SPACE);
+                sb.append(sortOrderField.getDirection());
+                sb.append(COMMA);
+                sb.append(SPACE);
             }
-            sb.append(SPACE);
-            sb.append(sortOrderField.getDirection());
-            sb.append(COMMA);
-            sb.append(SPACE);
         });
         // always sort with creation_date as a last order so that it supports cases where no values at all for the configured fields and also default fallback.
         return sb.append(CREATED_DATE + SPACE + SortDirection.fromOptionalString(metaData.getSortDirection())).toString();
     }
 
     private String getMataFieldName(String fieldName) {
         String metaFieldName = fieldName.startsWith("[") ? StringUtils.substringBetween(fieldName, "[", "]") : fieldName;
-        return CaseField.valueOf(metaFieldName).getDbColumnName();
+        String dbColumnName = null;
+        try {
+            dbColumnName = CaseField.valueOf(metaFieldName).getDbColumnName();
+        } catch (IllegalArgumentException iae) {
```

Comment:
You could let the exception propagate and not catch it, or rethrow it. Returning a null values will result in a more cryptic error later on.

Reply:
I followed your approach from here:
```    @Test
    public void sanitisesInputsCountQuery() {
        String evil = "foo');insert into case users values(1,2,3);--";
        when(authorisedCaseDefinitionDataService.getUserAuthorisedCaseStateIds("PROBATE", "TestAddressBookCase", CAN_READ))
            .thenReturn(asList(evil));

        when(userAuthorisation.getAccessLevel()).thenReturn(AccessLevel.GRANTED);
        when(userAuthorisation.getUserId()).thenReturn(evil);

        MetaData metadata = new MetaData("TestAddressBookCase", "PROBATE");
        final PaginatedSearchMetadata byMetaData = caseDetailsRepository.getPaginatedSearchMetadata(metadata, Maps.newHashMap());

        // If any input is not correctly sanitized it will cause an exception since query result structure will not be as hibernate expects.
        assertThat(byMetaData.getTotalResultsCount(), is(0));
    }

Where you also do not throw or propagate exception but do not append the section of sql query being built that contains a validation problem.

Reply:
The fix in that case was using OWASP option 1 - parameterised queries - it doesn't matter what the input is, it should still result in a valid query (that returns zero results).

In this case the column name is a parameter so we cannot use a parameterised query and must use OWASP option three - explicitly rejecting the bad input. So in this case I would design the test to expect that IllegalArgument exception and fail if not thrown.

Reply:
I would remove this exception handler completely and assert that the illegalargumentexception is thrown in your test